### PR TITLE
Consegna 20 Script 13/10

### PR DESCRIPTION
**Pull request by francosiNTTDATA**
In questa pr abbiamo sviluppato i 20 script

### CODE REVIEW

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 1]
```diff
@@ -27,6 +27,18 @@ public class RecapitiDestinatarioPage extends BasePage {
     @FindBy(xpath = "//button[@data-testid='codeCancelButton']")
     WebElement annullaButton;
 
+    @FindBy(xpath = "//button[@data-testid='add email']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:30]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 2]
```diff
@@ -27,6 +27,18 @@ public class RecapitiDestinatarioPage extends BasePage {
...
+    WebElement avvisamiMailButton;
+
+    @FindBy(xpath = "//button[@data-testid='add phone']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:33]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java [hunk 3]
```diff
@@ -123,6 +123,64 @@ public void clickOpzioneRevoca() {
...
+    }
+
+    public void clickRifiuta() {
+        try{
+            By rifiutaButtonBy = By.xpath("//li[contains(text(),'Rifiuta')]");
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(rifiutaButtonBy));
+            this.element(rifiutaButtonBy).click();
+            logger.info("Si visualizza il bottone rifiuta");
+        }catch(TimeoutException e){
+            logger.error("Non si clicca correttamente sul bottone rifiuta con errore"+e.getMessage());
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java]: insert : after string for divide string and message

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java [hunk 4]
```diff
@@ -123,6 +123,64 @@ public void clickOpzioneRevoca() {
...
+    }
+
+    public void clickRiufitaPopUp() {
+        try{
+            By rifiutaButtonPopUpBy = By.xpath("//button[contains(text(),'Rifiuta la delega')]");
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(rifiutaButtonPopUpBy));
+            this.element(rifiutaButtonPopUpBy).click();
+            logger.info("Si visualizza il bottone rifiuta nel pop-up");
+        }catch(TimeoutException e){
+            logger.error("Non si clicca correttamente sul bottone rifiuta pop-up con errore"+e.getMessage());
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java]: as before

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java [hunk 5]
```diff
@@ -123,6 +123,64 @@ public void clickOpzioneRevoca() {
...
+    }
+
+    public void clickAnnullaPopUp() {
+        try{
+            By annullaButtonPopUpBy = By.xpath("//button[contains(text(),'Annulla')]");
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(annullaButtonPopUpBy));
+            this.element(annullaButtonPopUpBy).click();
+            logger.info("Si visualizza il bottone rifiuta nel pop-up");
+        }catch(TimeoutException e){
+            logger.error("Non si clicca correttamente sul bottone rifiuta pop-up con errore"+e.getMessage());
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java]: as before

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java [hunk 6]
```diff
@@ -123,6 +123,64 @@ public void clickOpzioneRevoca() {
...
+        }
+    }
+
+    public boolean verificaEsistenzaDelega(String nome, String cognome) {
+        try {
+            By nomeDelegato = By.xpath("//div[@data-testid='delegators-wrapper']//div/p[contains(text(),'"+nome+" "+cognome+"')]");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(nomeDelegato));
+            return true;
+        }catch (TimeoutException | NoSuchElementException e){
+            return false;
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java:182]: insert log

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 7]
```diff
@@ -2,21 +2,44 @@
...
 public class ITuoiRecapitiPage extends BasePage {
     private static final Logger logger = LoggerFactory.getLogger("ITuoiRecapitiPage");
 
     @FindBy(xpath="//div[@data-testid='sideMenuItem-I tuoi recapiti']")
     WebElement iTuoiRecapitiButton;
 
+    @FindBy(id = "email")
+    WebElement emailInputField;
+
+    @FindBy(xpath = "//button[@data-testid='add email']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:22]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 8]
```diff
@@ -2,21 +2,44 @@
...
+    WebElement avvisamiViaEmailButton;
+
+    @FindBy(id = "phone")
+    WebElement phoneNumInputField;
+
+    @FindBy(xpath = "//button[@data-testid='add phone']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:28]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 9]
```diff
@@ -2,21 +2,44 @@
...
+    WebElement avvisamiViaSMSButton;
+
+    @FindBy(xpath = "//input[contains(@placeholder,'Il tuo indirizzo PEC')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 10]
```diff
@@ -2,21 +2,44 @@
...
+    WebElement inserimentoPecField;
+
+    @FindBy(xpath = "//button[@data-testid = 'addContact']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 11]
```diff
@@ -2,21 +2,44 @@
...
+    WebElement confermaButton;
+
+    @FindBy(xpath = "//button[contains(text(),'Annulla')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 12]
```diff
@@ -2,21 +2,44 @@
...
+    WebElement annullaButton;
+
+    @FindBy(xpath = "//button[contains(@data-testid,'codeConfirmButton')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 13]
```diff
@@ -38,4 +61,172 @@ public void waitLoadITuoiRecapitiPage() {
...
+        }
+    }
+
+    public void waitMessaggioErrore() {
+        try {
+            By messaggioErroreBy = By.xpath("//div[contains(@data-testid, 'errorAlert')]");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(messaggioErroreBy));
+            logger.info("Il messaggio di errore viene visualizzato correttamente");
+        }catch (TimeoutException e){
+            logger.error("Il messaggio di errore non viene visualizzato con errore: "+e.getMessage());
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: insert more explicit string

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 14]
```diff
@@ -38,4 +61,172 @@ public void waitLoadITuoiRecapitiPage() {
...
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(errorMessage));
+        return errorMessage.getText();
+    }
+
+    public boolean verificaButtoneConfermaDisabilitato(){
+        try{
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.confermaButton));
+            return Boolean.parseBoolean(this.confermaButton.getAttribute("disabled"));
+        }catch (NoSuchElementException | TimeoutException e){
+            return false;
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:123]: insert log

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 15]
```diff
@@ -38,4 +61,172 @@ public void waitLoadITuoiRecapitiPage() {
...
+    public void clickAvvisamiViaEmail(){
+        this.avvisamiViaEmailButton.click();
+    }
+
+    public boolean avvisamiViaEmailIsDisabled(){
+        try{
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.avvisamiViaEmailButton));
+            return Boolean.parseBoolean(this.avvisamiViaEmailButton.getAttribute("disabled"));
+        }catch (NoSuchElementException | TimeoutException e){
+            return false;
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:168]: as before

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java [hunk 16]
```diff
@@ -38,4 +61,172 @@ public void waitLoadITuoiRecapitiPage() {
...
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(errorMessage));
+        return errorMessage.getText();
+    }
+
+    public boolean avvisamiViaSMSIsDisabled(){
+        try{
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.avvisamiViaSMSButton));
+            return Boolean.parseBoolean(this.avvisamiViaSMSButton.getAttribute("disabled"));
+        }catch (NoSuchElementException | TimeoutException e){
+            return false;
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:218]: as before

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 17]
```diff
@@ -20,15 +20,45 @@ public class DeleghePGPagoPAPage extends BasePage {
...
     @FindBy(id = "taxId")
     WebElement cfTextField;
 
     @FindBy(xpath = "//button[@data-testid='confirmButton']")
     WebElement filtraButton;
 
+    @FindBy(xpath = "//button[contains(@data-testid,'codeConfirmButton')]")
+    WebElement accettaDelegaButton;
+
+    @FindBy(xpath = "//span[@data-testid='associate-group']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:32]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 18]
```diff
@@ -20,15 +20,45 @@ public class DeleghePGPagoPAPage extends BasePage {
...
+    WebElement assegnaGruppoRadioButton;
+
+    @FindBy(xpath = "//div[@role='dialog']//input[@id='groups']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:35]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 19]
```diff
@@ -20,15 +20,45 @@ public class DeleghePGPagoPAPage extends BasePage {
...
+    WebElement gruppoField;
+
+    @FindBy(id = "groups-option-0")
+    WebElement gruppoOption;
+
+    @FindBy(xpath = "//button[@data-testid='groupConfirmButton']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 20]
```diff
@@ -20,15 +20,45 @@ public class DeleghePGPagoPAPage extends BasePage {
...
+    WebElement confermaButton;
+
+    @FindBy(xpath = "//span[@data-testid='no-group']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:44]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 21]
```diff
@@ -20,15 +20,45 @@ public class DeleghePGPagoPAPage extends BasePage {
...
+    WebElement nonGruppoRadioButton;
+
+    @FindBy(xpath = "//li[contains(text(),'Rifiuta')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 22]
```diff
@@ -20,15 +20,45 @@ public class DeleghePGPagoPAPage extends BasePage {
...
+    WebElement opzioneRevoca;
+
+    @FindBy(xpath = "//button[contains(@class,'MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 23]
```diff
@@ -20,15 +20,45 @@ public class DeleghePGPagoPAPage extends BasePage {
...
+    WebElement rifiutaButton;
+
+    @FindBy(xpath = "//li[contains(text(),'Modifica')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 24]
```diff
@@ -126,4 +139,104 @@ public boolean controlloDelegaRestituita(String ragioneSociale) {
...
+        this.opzioneRevoca.click();
+    }
+
+    public void clickBottoneRifiuta() {
+        this.rifiutaButton.click();
+    }
+
+    public void waitLoadPopUpRevoca() {
+        try {
+            By revocaPopUpBy = By.xpath("//div[@aria-labelledby='responsive-dialog-title']");
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:200]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java [hunk 25]
```diff
@@ -126,4 +139,104 @@ public boolean controlloDelegaRestituita(String ragioneSociale) {
...
+        }
+    }
+
+    public boolean verificaPresenzaGruppo(String ragioneSociale) {
+        try {
+            By gruppoBy = By.xpath("//td[@role='cell' and div/p[contains(text(),'"+ragioneSociale+"')]]/following-sibling::td[@role='cell']//span[contains(text(),'Test gruppi')]");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(gruppoBy));
+            return true;
+        }catch (TimeoutException e) {
+            return false;
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:232]: insert log

File: src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java [hunk 26]
```diff
@@ -20,6 +20,14 @@ public class DelegatiImpresaSection extends BasePage {
 
     @FindBy(xpath = "//li[contains(@tabindex,'0')]")
     WebElement mostraCodiceOption;
+    @FindBy(xpath = "//li[contains(text(),'Revoca')]")
+    WebElement revocaMenuButton;
+
+    @FindBy(xpath = "//button[@data-testid='dialogAction' and text()='Revoca la delega']")
+    WebElement revocaButton;
+
+    @FindBy(xpath = "//button[contains(text(),'Annulla')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java [hunk 27]
```diff
@@ -20,6 +20,14 @@ public class DelegatiImpresaSection extends BasePage {
 
     @FindBy(xpath = "//li[contains(@tabindex,'0')]")
     WebElement mostraCodiceOption;
+    @FindBy(xpath = "//li[contains(text(),'Revoca')]")
+    WebElement revocaMenuButton;
+
+    @FindBy(xpath = "//button[@data-testid='dialogAction' and text()='Revoca la delega']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java [hunk 28]
```diff
@@ -20,6 +20,14 @@ public class DelegatiImpresaSection extends BasePage {
 
     @FindBy(xpath = "//li[contains(@tabindex,'0')]")
     WebElement mostraCodiceOption;
+    @FindBy(xpath = "//li[contains(text(),'Revoca')]")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java]: report xpath for change in id

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java (hunk 1 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:30]: done

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java (hunk 2 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:33]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java (hunk 3 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java (hunk 4 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java (hunk 5 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 7 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:22]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 8 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:28]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 9 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 10 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 11 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 13 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: we cancel this method becasue not used

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 17 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:32]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 19 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 20 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:44]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 18 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:35]: done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 21 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: done and alerady change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 22 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 23 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 24 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:200]: done

File: src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java (hunk 26 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java (hunk 27 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java (hunk 28 above)
Comment (delfinoroNTTDATA) [reply src/main/java/it/pn/frontend/e2e/section/destinatario/personaGiuridica/DelegatiImpresaSection.java]: done and already change

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java (hunk 6 above)
Comment (francosiNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/DeleghePage.java:182]: Done

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 12 above)
Comment (francosiNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java]: Done and already changed

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 15 above)
Comment (francosiNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:168]: Logger already present in the step that calls this method

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 14 above)
Comment (francosiNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:123]: Logger already present in the step that calls this method

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java (hunk 16 above)
Comment (francosiNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java:218]: Logger already present in the step that calls this method

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java (hunk 25 above)
Comment (francosiNTTDATA) [reply src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java:232]: Logger already present in the step that calls this method

**Review by leleOFA — APPROVED** [review]

**Pull request merged**